driver/resource_pool.go

```diff
@@ -19,15 +20,44 @@ func (d *Driver) NewResourcePool(ref *types.ManagedObjectReference) *ResourcePoo
 	}
 }
 
-func (d *Driver) FindResourcePool(host string, name string) (*ResourcePool, error) {
-	p, err := d.finder.ResourcePool(d.ctx, fmt.Sprintf("/%v/host/%v/Resources/%v", d.datacenter.Name(), host, name))
-	if err != nil {
-		return nil, err
+func (d *Driver) FindResourcePool(name string) (*ResourcePool, error) {
+	t := d.client.ServiceContent.About.ApiType
+	switch t {
+	case "HostAgent":
+		ddc, err := d.finder.DefaultDatacenter(d.ctx)
+		if err != nil {
+			return nil, err
+		}
+		d.finder.SetDatacenter(ddc)
+		p, err := d.finder.DefaultResourcePool(d.ctx)
+		if err != nil {
+			return nil, err
+		}
+		return &ResourcePool{
+			pool:   p,
+			driver: d,
+		}, nil
+	case "VirtualCenter":
+		if name != "" {
+			p, err := d.finder.ResourcePool(d.ctx, name)
+			if err != nil {
+				return nil, err
+			}
+			return &ResourcePool{
+				pool:   p,
+				driver: d,
+			}, nil
+		}
+		p, err := d.finder.DefaultResourcePool(d.ctx)
```

Comment:
I have multiple resource pools , so it fails with `Build 'vsphere-iso' errored: error creating vm: default resource pool resolves to multiple instances, please specify`

Reply:
As explained in the previous comment, if you have more than one root `resource_pool` or `resource_pools` with the same name (and declare them on the config only by the name, not the full path), this will happen. This just means that you must be a little more specific and setting the complete path to the desired `resource_pool` would solve the issue.